core/chains/evm/client/chain_client.go

```diff
@@ -294,53 +365,78 @@ func (c *chainClient) SendTransactionReturnCode(ctx context.Context, tx *types.T
 }
 
 func (c *chainClient) SequenceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (evmtypes.Nonce, error) {
-	return c.multiNode.SequenceAt(ctx, account, blockNumber)
+	rpc, err := c.multiNode.SelectRPC()
+	if err != nil {
+		return 0, err
+	}
+	return rpc.SequenceAt(ctx, account, blockNumber)
 }
 
 func (c *chainClient) SubscribeFilterLogs(ctx context.Context, q ethereum.FilterQuery, ch chan<- types.Log) (s ethereum.Subscription, err error) {
-	rpc, err := c.multiNode.SelectNodeRPC()
+	rpc, err := c.multiNode.SelectRPC()
 	if err != nil {
 		return s, err
 	}
 	return rpc.SubscribeFilterLogs(ctx, q, ch)
 }
 
-func (c *chainClient) SubscribeNewHead(ctx context.Context, ch chan<- *evmtypes.Head) (ethereum.Subscription, error) {
-	csf := newChainIDSubForwarder(c.ConfiguredChainID(), ch)
-	err := csf.start(c.multiNode.Subscribe(ctx, csf.srcCh, "newHeads"))
+func (c *chainClient) SubscribeNewHead(ctx context.Context) (<-chan *evmtypes.Head, ethereum.Subscription, error) {
+	rpc, err := c.multiNode.SelectRPC()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	return csf, nil
+
+	ch, sub, err := rpc.SubscribeToHeads(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	chainID, err := c.ChainID()
```

Comment:
Make sure we are using configured value here. MultiNode is expected to guarantee that it's the same as provided by RPC

Reply:
Sounds good! We can actually just remove this `ChainID()` method from ChainClient and use configured chain id.